### ISSUE TITLE
Regression - Fail to download exchange rates

### ISSUE THREAD

**Issue by qnswirlds** [hashgraph/hedera-services#897]
```
2020-12-24 01:39:03.082 ERROR  204  HapiApiSpec - Fees failed to initialize!
java.lang.IllegalStateException: Could not download '0.0.112' final status INVALID_SIGNATURE
```
Test clients fail with this error in many regression tests.

Link for the failed test : https://hedera-hashgraph.slack.com/archives/CKWHL8R9A/p1609154833151900

**Comment by Neeharika-Sompalli** [comment]
@anighanta  One other HTS test failed with a NPE while calculating Fees for Token Create. It might be related to the same issue. 
The link for failed test is : https://hedera-hashgraph.slack.com/archives/CKWHL8R9A/p1609137037137300

Error is as follows:
```2020-12-28 05:44:30.269 WARN   83   FeeCalculator - Unable to calculate fee for op TokenCreate, using max fee!
java.lang.NullPointerException: null
	at com.hederahashgraph.fee.FeeBuilder.getFeeObject(FeeBuilder.java:130) ~[SuiteRunner.jar:?]
	at com.hederahashgraph.fee.FeeBuilder.getTotalFeeforRequest(FeeBuilder.java:122) ~[SuiteRunner.jar:?]
	at com.hedera.services.bdd.spec.fees.FeeCalculator.forOp(FeeCalculator.java:80) ~[SuiteRunner.jar:?]
	at com.hedera.services.bdd.spec.fees.FeeCalculator.forActivityBasedOp(FeeCalculator.java:98) ~[SuiteRunner.jar:?]
	at com.hedera.services.bdd.spec.transactions.token.HapiTokenCreate.feeFor(HapiTokenCreate.java:177) ~[SuiteRunner.jar:?]
	at com.hedera.services.bdd.spec.HapiSpecOperation.finalizedTxn(HapiSpecOperation.java:284) ~[SuiteRunner.jar:?]
	at com.hedera.services.bdd.spec.HapiSpecOperation.finalizedTxn(HapiSpecOperation.java:257) ~[SuiteRunner.jar:?]
	at com.hedera.services.bdd.spec.transactions.HapiTxnOp.submitOp(HapiTxnOp.java:137) ~[SuiteRunner.jar:?]
	at com.hedera.services.bdd.spec.HapiSpecOperation.execFor(HapiSpecOperation.java:179) ~[SuiteRunner.jar:?]
	at com.hedera.services.bdd.spec.utilops.CustomSpecAssert.allRunFor(CustomSpecAssert.java:38) ~[SuiteRunner.jar:?]
	at com.hedera.services.bdd.spec.utilops.CustomSpecAssert.allRunFor(CustomSpecAssert.java:46) ~[SuiteRunner.jar:?]
	at com.hedera.services.bdd.spec.utilops.ProviderRun.submitOp(ProviderRun.java:101) ~[SuiteRunner.jar:?]
	at com.hedera.services.bdd.spec.HapiSpecOperation.execFor(HapiSpecOperation.java:179) ~[SuiteRunner.jar:?]
	at com.hedera.services.bdd.spec.HapiApiSpec.exec(HapiApiSpec.java:251) ~[SuiteRunner.jar:?]
	at com.hedera.services.bdd.spec.HapiApiSpec.run(HapiApiSpec.java:190) ~[SuiteRunner.jar:?]
	at java.util.AbstractList$RandomAccessSpliterator.forEachRemaining(AbstractList.java:720) ~[?:?]
	at java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:658) ~[?:?]
	at com.hedera.services.bdd.suites.HapiApiSuite.runSync(HapiApiSuite.java:250) ~[SuiteRunner.jar:?]
	at com.hedera.services.bdd.suites.HapiApiSuite.runSuite(HapiApiSuite.java:141) ~[SuiteRunner.jar:?]
	at com.hedera.services.bdd.suites.HapiApiSuite.runSuiteSync(HapiApiSuite.java:134) ~[SuiteRunner.jar:?]
	at com.hedera.services.bdd.suites.SuiteRunner.lambda$runSuitesSync$15(SuiteRunner.java:504) ~[SuiteRunner.jar:?]
	at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:176) ~[?:?]
	at java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:948) ~[?:?]
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484) ~[?:?]
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474) ~[?:?]
	at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913) ~[?:?]
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234) ~[?:?]
	at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578) ~[?:?]
	at com.hedera.services.bdd.suites.SuiteRunner.runSuitesSync(SuiteRunner.java:505) ~[SuiteRunner.jar:?]
	at com.hedera.services.bdd.suites.SuiteRunner.lambda$runTargetCategories$10(SuiteRunner.java:479) ~[SuiteRunner.jar:?]
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195) [?:?]
	at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1654) [?:?]
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484) [?:?]
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474) [?:?]
	at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913) [?:?]
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234) [?:?]
	at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578) [?:?]
	at com.hedera.services.bdd.suites.SuiteRunner.runTargetCategories(SuiteRunner.java:479) [SuiteRunner.jar:?]
	at com.hedera.services.bdd.suites.SuiteRunner.runCategories(SuiteRunner.java:448) [SuiteRunner.jar:?]
	at com.hedera.services.bdd.suites.SuiteRunner.main(SuiteRunner.java:369) [SuiteRunner.jar:?]
```

**Comment by anighanta** [comment]
@Neeharika-Sompalli  This shouldn't be happening frequently. It looks like the fee schedule might not have been downloaded fully. I don't this this has any relation with the Exchange rates not being downloaded. This test shows that the exchanges rates  are successfully downloaded and parsed. investigating more on the exchange rates download failed because of `INVALID_SIGNATURE` as in the original issue.

**Comment by anighanta** [comment]
Error `INVALID_SIGNATURE` is only happening on GCP tests that start from a saved state. On further digging we found out that link to the saved state file in s3 bucket is being converted into a link to Google Cloud Platform Storage link, making a copy of the bucket `services-regression-jrs-files` from s3 in to GCP Storage fixed this issue. 

Closing this now. 

Test Results : https://hedera-hashgraph.slack.com/archives/C018CBQBTGB/p1609198144335900
